internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -201,6 +212,71 @@ func (c DockerCmdClient) Push(ctx context.Context, uri string, w io.Writer, tags
 	return parts[1], nil
 }
 
+func (in *RunOptions) generateRunArguments() []string {
+	args := []string{"run"}
+
+	//Add container name option.
+	if in.ContainerName != "" {
+		args = append(args, "--name", in.ContainerName)
+	}
+
+	// Add port mappings.
+	for hostPort, containerPort := range in.ContainerPorts {
+		args = append(args, "--publish", fmt.Sprintf("%s:%s", hostPort, containerPort))
+	}
+
+	// Add network option if it's not a "pause" container.
+	if !strings.HasPrefix(in.ContainerName, "pause") {
+		args = append(args, "--network", fmt.Sprintf("container:%s", in.ContainerNetwork))
+	}
+
+	//Add secrets as environment variables.
+	for key, value := range in.Secrets {
+		args = append(args, "--env", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add environment variables.
+	for key, value := range in.EnvVars {
+		args = append(args, "--env", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add the image name.
+	if in.ImageURI != "" {
+		args = append(args, in.ImageURI)
+	}
+
+	if in.Command != nil && len(in.Command) > 0 {
+		args = append(args, in.Command...)
+	}
+	return args
+}
+
+// Run runs a Docker container with the sepcified options.
+func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
+	args := options.generateRunArguments()
+
+	//Execute the Docker run command.
+	if err := c.runner.RunWithContext(ctx, "docker", args); err != nil {
```

Comment:
nit nit nit:
```suggestion
	if err := c.runner.RunWithContext(ctx, "docker", options.generateRunArguments()); err != nil {
```
(feel free to address this nit later if you are currently using the `args` variable for debugging)